### ISSUE TITLE
Question: How to check if alert is present

### ISSUE THREAD

**Issue by valfirst** [MachinePublishers/jBrowserDriver#147]
CrhomeDriver, FirefoxDriver and etc. throw `org.openqa.selenium.NoAlertPresentException` at `webDriver.switchTo().alert()` when no alert is present. But jBrowserDriver always returns an object.
Is it an issue? If it's not, how can I identify whether alert is present or not using jBrowserDriver?

**Comment by hollingsworthd** [comment]
Yes, it's a bug that it doesn't follow the [spec](http://machinepublishers.github.io/jBrowserDriver/org/openqa/selenium/WebDriver.TargetLocator.html#alert--).

Best workaround would be to call `webDriver.switchTo().alert().getText()` and check for null... it will wait for an alert to be shown until the script timeout is reached. Potentially you will not be able to differentiate between an actual empty alert text and an alert not being shown.

**Comment by hollingsworthd** [comment]
For the workaround, set driver.manage().timeouts().setScriptTimeout(..)
